### PR TITLE
Sulfuric Acid Dispenser

### PR DESCRIPTION
**Pull request by Apogee-dev**
sacidtank2

Adds a wall-mounted sulphuric acid dispenser to the game, and places dispensers in Robotics and Engineering.

:cl:
 * rscadd: Added sulphuric acid dispenser to Robotics and Engineering.

### CODE REVIEW

**Comment by Eneocho** [comment]
Did you remove the RnD sacid beaker?

**Comment by Apogee-dev** [comment]
> Did you remove the RnD sacid beaker?

No. They still need a beaker to fill the imprinter with, after all.

**Comment by Eneocho** [comment]
That's what the autolathe is for, you can make a large beaker for 2000 glass, that's less than a sheetworth.
Or make a bluespace beaker on the protolathe (You can make them before you reach the circuits part of the RnD checklist).

File: code/modules/reagents/reagent_dispenser.dm [hunk 1]
```diff
@@ -317,6 +317,37 @@
 	. = ..()
 	reagents.add_reagent(VIRUSFOOD, 1000)
 
+
+/obj/structure/reagent_dispensers/sulfacid
+	name = "\improper Sulfuric Acid Tank"
+	desc = "A tank filled with sulfuric acid."
```
Comment (Eneocho) [inline code/modules/reagents/reagent_dispenser.dm:323]:
```suggestion
	name = "\improper Sulfuric Acid Dispenser"
	desc = "A dispenser of sulfuric acid."
```

**Comment by Apogee-dev** [comment]
> That's what the autolathe is for, you can make a large beaker for 2000 glass, that's less than a sheetworth.
> Or make a bluespace beaker on the protolathe (You can make them before you reach the circuits part of the RnD checklist).

I'll just replace them with empty beakers, I suppose.

File: code/modules/reagents/reagent_dispenser.dm [hunk 2]
```diff
@@ -317,6 +317,37 @@
 	. = ..()
 	reagents.add_reagent(VIRUSFOOD, 1000)
 
+
+/obj/structure/reagent_dispensers/sulfacid
+	name = "\improper Sulfuric Acid Dispenser"
+	desc = "A dispender of sulfuric acid."
```
Comment (Eneocho) [inline code/modules/reagents/reagent_dispenser.dm]:
```suggestion
	name = "\improper Sulphuric Acid Dispenser"
	desc = "A dispenser of sulphuric acid."
```
Sulphuric for consistency.

Comment (Apogee-dev) [reply code/modules/reagents/reagent_dispenser.dm]: Sweet jesus i'm an idiot

**Comment by Eneocho** [comment]
Come on man. The reagent name is "sulphuric acid". Replace "sulfuric" with "sulphuric" to have consistency.

**Comment by Apogee-dev** [comment]
> Come on man. The reagent name is "sulphuric acid". Replace "sulfuric" with "sulphuric" to have consistency.

this is why i need to stop coding after midnight

**Comment by DeityLink** [comment]
I'm fine with science having them since they have chem dispensers on the roid anyway, but if even engineering has them then no one will ever pester Chemistry for sulph acid ever again and that makes me sad (not to mention engineers don't need sulph acid that often)

**Comment by SonixApache** [comment]
and further down the no-interaction streamlined game slope we go

**Comment by gurfan** [comment]
I don't really see the point of this in science

**Comment by Durgen** [comment]
i'd rather see this as a gimmick on one map

**Comment by Blithering** [comment]
Finally, greys get their own sinks.

**Comment by DeityLink** [comment]
> 
> 
> I don't really see the point of this in science

R&D needs this dispenser the most out of any job to be fair.

**Comment by gurfan** [comment]
they would use it the most, but I doubt that they *need* it when building a chemistry dispenser is still part of the usual r&d routine

**Comment by Kurfursten** [comment]
I don't like reducing interdepartment cooperation.

Engineers get enough starting sacid to handle all but the biggest soldering jobs, and I've *never* had a problem as an engineering main getting a refill at medbay. R&D deserves to do the walk of shame if they spill their acid before getting a dispenser up.

**Comment by Apogee-dev** [comment]
> they would use it the most, but I doubt that they _need_ it when building a chemistry dispenser is still part of the usual r&d routine

Building a chemistry dispenser is only part of the usual RnD routine because they need a convenient source of sulfacid for circuits so they can finish RnD. Half the point of this is to take away that reason so RnD doesn't strictly need to build a dispenser every shift; usually, the _only_ thing they use it for is acid. The other side of the coin is that I'm leery of leaving a dispenser in robotics without a matching one in RnD, because the scientists will probably just break into robotics to use it.

> Engineers get enough starting sacid to handle all but the biggest soldering jobs, and I've never had a problem as an engineering main getting a refill at medbay.

Engineers do get enough starting sacid to handle regular jobs, true. When I first brought this idea up I got several stories about the engineers running out during mass emaggings (four and a half solders' worth apparently runs out faster than you'd think), and the mechanics also have a habit of pilfering Engineering's acid supply for their imprinter, but as you mentioned engineers roam a lot and usually don't have any issue getting refills. I'm open to removing either the engineering or mechanics dispenser, but I don't think removing both would be a good idea due to the aforementioned mechanics-stealing-acid thing.

**Comment by TurboGay** [comment]
> 
> 
> I don't like reducing interdepartment cooperation.
> 
> Engineers get enough starting sacid to handle all but the biggest soldering jobs, and I've _never_ had a problem as an engineering main getting a refill at medbay. R&D deserves to do the walk of shame if they spill their acid before getting a dispenser up.

Sci starts with a dispenser on literally every map so I really don't see the issue with it

**Comment by Apogee-dev** [comment]
> Sci starts with a dispenser on literally every map so I really don't see the issue with it

This is true. The only thing that keeps them from using it more often is that on most maps it requires a shuttle ride (except Roid. Scientists don't use that one because not that many remember it's there.)

**Comment by Apogee-dev** [comment]
> I don't like reducing interdepartment cooperation.

also just to touch on this statement again, the only one that could be said to reduce interdepartment cooperation here is engineering. RnD builds their own dispenser every shift and has a backup on the outpost, Robotics generally relies on RnD for acid, mechanics usually scan a dispenser in chem at roundstart and set up their own and/or steal acid bottles from engineering.

**Comment by Apogee-dev** [comment]
Gonna nix the dispensers in RnD and Mechanics and see where this goes. Mechanics can fill up at engineering, engineers can use their dispenser if mechanics steal their bottles. RnD has the usual route and roboticists can get acid without breaking into rnd.

**Comment by Eneocho** [comment]
> 
> 
> Gonna nix the dispensers in RnD and Mechanics and see where this goes. Mechanics can fill up at engineering, engineers can use their dispenser if mechanics steal their bottles. RnD has the usual route and roboticists can get acid without breaking into rnd.

I'd close this PR and make a new one so people can see the changes.

**Comment by Apogee-dev** [comment]
> I'd close this PR and make a new one so people can see the changes.

I updated the description and everybody who's pitched in should get notifications about the commit.

**Comment by Eneocho** [comment]
> 
> 
> > I'd close this PR and make a new one so people can see the changes.
> 
> I updated the description and everybody who's pitched in should get notifications about the commit.

Good idea in theory, doesn't really work in practice. Also, you forgot to change the changelog.

**Comment by Apogee-dev** [comment]
If this one flops, it flops. I'll probably make a Take 2 that just makes draggable sulfacid tanks available in cargo, none mapped in at roundstart. that way you can take a trip to cargo instead of chem if you need a personal supply.

**Comment by Apogee-dev** [comment]
I'm gonna close this one early. Opinion on this specific implementation seems to be split at best and, at this point, I don't think I'm satisfied with this implementation either- I'll put up an alternative that should hopefully prove more agreeable at a later date.

**Pull request closed without merging**